### PR TITLE
bring back support for `favicon.ico`

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/20487

## What I did

- resolve the favicon path in a preset-property
- pick the first one, if any is found, warn user, if we found multiple
- make the link tag injection work for both svg and ico

### CODE REVIEW

**Comment by ndelangen** [comment]
I'd like to give this a manual test before merging

**Review by valentinpalkovic — APPROVED** [review]

File: code/lib/core-server/src/presets/common-preset.ts [hunk 1]
```diff
@@ -10,6 +15,67 @@ import type {
   StorybookConfig,
 } from '@storybook/types';
 import { loadCsf } from '@storybook/csf-tools';
+import { join } from 'path';
+import { dedent } from 'ts-dedent';
+import { parseStaticDir } from '../utils/server-statics';
+
+const defaultFavIcon = require.resolve('@storybook/core-server/public/favicon.svg');
```
Comment (shilman) [inline code/lib/core-server/src/presets/common-preset.ts:22]:
```suggestion
const defaultFavicon = require.resolve('@storybook/core-server/public/favicon.svg');
```

Comment (shilman) [reply code/lib/core-server/src/presets/common-preset.ts:22]: Capitalization consistency 😄

File: code/lib/core-server/src/presets/common-preset.ts [hunk 2]
```diff
@@ -10,6 +15,67 @@ import type {
...
+
+export const favicon = async (value: string, options: Options) => {
```
Comment (shilman) [inline code/lib/core-server/src/presets/common-preset.ts:24]: Can we please add unit tests for this?

**Review by shilman — CHANGES_REQUESTED** [review]
Generally LGTM!

File: code/lib/core-server/src/presets/common-preset.ts (hunk 2 above)
Comment (ndelangen) [reply code/lib/core-server/src/presets/common-preset.ts]: yes!

**Pull request merged**